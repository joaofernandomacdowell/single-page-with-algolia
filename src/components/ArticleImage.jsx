import React from 'react';

const ArticleImage = ({ hit }) => {
  const { imageUrl, title, url } = hit;
  return (
    <a className="zoom-article-image__wrapper" href={url}>
      <figure className="zoom-article-image__figure">
        <img className="zoom-article-image__img"
          src={imageUrl}
          alt={title}
          title={title}
          loading="lazy"
        />
      </figure>
    </a>
  );
}

export default ArticleImage;