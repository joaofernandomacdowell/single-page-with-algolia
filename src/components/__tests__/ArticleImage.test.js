import React from 'react'
import { mount } from 'enzyme';

import mockApiResponse from '../../helper';
import ArticleImage from '../ArticleImage';


describe('ArticleImage component', () => {
  const props = {
    hit: mockApiResponse.hits[0]
  };
  const wrapper = mount(<ArticleImage {...props} />);

  test('match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('have anchor container with the correct href', () => {
    const anchor = wrapper.find('.zoom-article-image__wrapper');

    expect(anchor.props().href).toBe(mockApiResponse.hits[0].url);
  });

  test('have image with required props and values', () => {
    const image = wrapper.find('.zoom-article-image__img');

    expect(image.props().src).toBe(mockApiResponse.hits[0].imageUrl);
    expect(image.props().title).toBe(mockApiResponse.hits[0].title);
    expect(image.props().alt).toBe(mockApiResponse.hits[0].title);
  });

  test('lazy loading is seted', () => {
    const image = wrapper.find('.zoom-article-image__img');

    expect(image.props().loading).toBe('lazy');
  });
});