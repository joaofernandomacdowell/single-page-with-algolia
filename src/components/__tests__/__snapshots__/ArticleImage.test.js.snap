// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleImage component match the snapshot 1`] = `ReactWrapper {}`;
